src/index.ts

```diff
@@ -54,8 +54,17 @@ export interface BasicSpan {
  * @param text Any valid USVString with words to split.
  */
 export function split(text: string): string[] {
-  let spans = Array.from(findSpans(text));
-  return spans.map(span => span.text).filter(isNonSpace);
+  const spans: string[] = [];
+  
+  const boundaries = findBoundaries(text)
+  let prev = boundaries.next().value;
+  if(prev === undefined) return [];
```

Comment:
Please match the style guides of the repo before opening the PR

```suggestion
  if (prev === undefined) return [];
```